src/biogeophys/SoilHydrologyMod.F90

```diff
@@ -1705,89 +1709,77 @@ subroutine PerchedLateralFlow(bounds, num_hydrologyc, filter_hydrologyc, &
 
        dtime = get_step_size_real()
 
-       ! Compute ice fraction in each layer
-
-       do j = 1,nlevsoi
-          do fc = 1, num_hydrologyc
-             c = filter_hydrologyc(fc)
-             dzmm(c,j) = dz(c,j)*1.e3_r8
+       ! locate frost table and perched water table
+       do fc = 1, num_hydrologyc
+          c = filter_hydrologyc(fc)
+          k_frost(c) = nbedrock(c)
+          k_perch(c) = nbedrock(c)
+          do k = 1, nbedrock(c)
+             if (frost_table(c) >= zi(c,k-1) .and. frost_table(c) < zi(c,k)) then
+                k_frost(c) = k
+                exit
+             endif
+          enddo
 
-             vol_ice = min(watsat(c,j), h2osoi_ice(c,j)/(dz(c,j)*denice))
-             icefrac(c,j) = min(1._r8,vol_ice/watsat(c,j))          
-          end do
-       end do
+          do k = 1, nbedrock(c)
+             if (zwt_perched(c) >= zi(c,k-1) .and. zwt_perched(c) < zi(c,k)) then
+                k_perch(c) = k
+                exit
+             endif
+          enddo
+       enddo
 
        ! compute drainage from perched saturated region
        do fc = 1, num_hydrologyc
           c = filter_hydrologyc(fc)
 
           qflx_drain_perched(c) = 0._r8
+          if (frost_table(c) > zwt_perched(c)) then
 
-          if ((frost_table(c) > zwt_perched(c)) .and. origflag == 0) then
```

Comment:
- [ ] Just double-checking the removal of origflag from this conditional, and the removal of `if(origflag == 1) qflx_drain_perched(c) = 0._r8` below: I know we have talked about dropping support for origflag = 1. But since we haven't yet, I wanted to double-check: is it okay to just do this block of code even with origflag = 1, or is now the time to go ahead and remove origflag entirely?